backend/infrahub/computed_attribute/tasks.py

```diff
@@ -46,6 +54,35 @@
 """
 
 
+async def wait_until_flow_runs_are_completed(
+    flow_run_ids: list[UUID], interval: int = 2, timeout: int | None = 3600
+) -> list[UUID]:
+    pending_flow_run_ids: list[UUID] = flow_run_ids
+    failed_flow_run_ids: list[UUID] = []
+
+    ACTIVE_FLOW_RUN_STATES = [StateType.RUNNING, StateType.PENDING, StateType.SCHEDULED]
+    FAILED_FLOW_RUN_STATES = [StateType.FAILED, StateType.CANCELLED, StateType.CRASHED]
+
+    start_time = time.time()
+
+    async with get_client(sync_client=False) as prefect_client:
+        while pending_flow_run_ids:
+            flow_runs = await prefect_client.read_flow_runs(
+                flow_run_filter=FlowRunFilter(id=FlowRunFilterId(any_=pending_flow_run_ids))
+            )
+            pending_flow_run_ids = [
+                flow_run.id for flow_run in flow_runs if flow_run.state_type in ACTIVE_FLOW_RUN_STATES
+            ]
+            failed_flow_run_ids.extend(
+                [flow_run.id for flow_run in flow_runs if flow_run.state_type in FAILED_FLOW_RUN_STATES]
+            )
+            await asyncio.sleep(interval)
```

Comment:
I think we want an `if pending_flow_run_ids:` here so that we'd only add an additional 2 second sleep if there are still pending flow runs.